### PR TITLE
Roll call's attendee UI

### PR DESCRIPTION
**Pull request by Kar1213**
This code uses the new go backend code from PR #294 which needs to be run for the app to work.
These are the new functionalities:
- The LAO properties show the channel id as a QR code so that users can subscribe
- A fragment for attendees to display their public key to join roll calls
- Updating the LAO name (expect for a detail, see issue #313)
- A confirmation popup when an attendee is successfully scanned

Overall, a first version of the complete process of running a roll call should be operational (without wallet).

NEW updates to add witness_signatures field in the json of the messages sent to the backend:
- PublicKeySignaturePair.java: add getters for base64 encoded fields
- JsonMessageGeneralSerializer.java: correct the way witness_signatures field is added to the messages
- MessageGeneral.java: initialize signatures list with empty array

### CODE REVIEW

**Review by celinecamacho — COMMENTED** [review]
There are some go files on this PR, is this intended?

**Comment by gnarula** [comment]
> There are some go files on this PR, is this intended?

We can either

* Remove that merge commit and process this PR independently
* Or we wait for #294 to be merge

**Review by Walfar — COMMENTED** [review]
Nice job ! I'd only suggest to divide your changes into differents PRs next time (e.g the witness signatures), so that it's easier to review and we can merge sooner :) 
It would also be nice to add a few doc there and there, but since we are in the emergency of Wednesday's POP party, I guess this can wait !
Also, I will have to test your code on my side (tomorrow) before approving, to make sure that everything is working fine.

File: fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/LaoDetailViewModel.java
Comment (Kar1213) [reply fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/LaoDetailViewModel.java:416]: Good point I will add it.

File: fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/fragments/RollCallDetailFragment.java
Comment (Kar1213) [reply fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/fragments/RollCallDetailFragment.java:28]: Right this is better thanks.

File: fe2-android/app/src/main/java/com/github/dedis/student20_pop/utility/json/JsonMessageGeneralSerializer.java
Comment (Kar1213) [reply fe2-android/app/src/main/java/com/github/dedis/student20_pop/utility/json/JsonMessageGeneralSerializer.java:48]: Yes, the backend team told me they added new constraints on this field.

**Comment by Kar1213** [comment]
Thanks for your review Victor I made the changes you mentioned. Regarding the Base64Url encoding I changed the ones that are not in your PR (where I used Base64 in new places). Normally the whole codebase should use this encoding after merging both PRs.

I also made a new fix regarding how the wallet is accessed: for the wallet we need to use the same (lao id, rollcall id) pair when retrieving a keyset.
The issue was that the roll call ids are changed when switching from one state to the other (created, opened, closed) so we would get different keysets for the same roll call. Similarly for lao ids where it changes for example when updating the lao name. Therefore I added a new field called persistentId in the RollCall.java class which is the id given at creation (done in handleCreateLao() in LaoRepository.java). For the lao id, I used the channel id since it is never modified after creation.

**Review by celinecamacho — COMMENTED** [review]
Nice job 👍 I have some comments on the implementation, mostly small details.

File: fe2-android/app/src/main/java/com/github/dedis/student20_pop/utility/json/JsonMessageGeneralSerializer.java
Comment (Kar1213) [reply fe2-android/app/src/main/java/com/github/dedis/student20_pop/utility/json/JsonMessageGeneralSerializer.java:28]: Because SonarCould was complaining that string "signature" appears 4 times in the file

File: fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/LaoDetailViewModel.java
Comment (Walfar) [reply fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/LaoDetailViewModel.java:541]: Thank you, its much clearer now :)

**Comment by Kar1213** [comment]
Thanks for your review Céline I made the changes you pointed out. I also changed back the encoding to base64 to avoid new problems we didn't expect.

**Review by maximelaval — APPROVED** [review]
Great Work ! Everything seems good after testing the application with the backend. I was just wondering if it's normal if the fragment to ask the permission for the camera doesn't display for the first time I try to open a Roll Call ? Is it enabled by default ?

**Comment by Kar1213** [comment]
Thank you Maxime. For the camera permission I am not 100% sure (this was already implemented) but I think you just need to give the permission the first time you access the app and then it remembers it in the app context for later.

**Pull request merged**
